packages/data-context/src/actions/ProjectActions.ts

```diff
@@ -495,6 +501,15 @@ export class ProjectActions {
       }
     })
 
+    const specPattern = await this.ctx.project.specPatternForTestingType(project.projectRoot, 'e2e')
+
+    if (!specPattern) {
```

Comment:
I'm guessing this can happen if the user has configured it to an invalid value? Seems like it should be handled by config validation instead.

Reply:
Yep, I just put this here to appease TypeScript. We need to update the configuration validation a bunch, at this point in the code we should have validating everything already. A lot of work around config is [happening here](https://github.com/cypress-io/cypress/pull/19347) and in another PR. I didn't add any validation here other than some simple inline checks in order to keep this PR "small" (if you can really call 100 files small...)